test_path.py

```diff
@@ -937,5 +937,21 @@ def test_exception_in_context(self, tmpdir):
         assert not 'Lorem' in data
         assert 'lazy dog' in data
 
+
+@pytest.mark.skipif(sys.platform.startswith('win'),
+                    reason="requires non-Windows (POSIX)")
+def test_posix_url():
+    assert Path('/var/log/xyzzy').url() == 'file:///var/log/xyzzy'
+    assert Path('xyzzy').url() == 'file:///xyzzy'
+
+
+@pytest.mark.skipif(not sys.platform.startswith('win'),
+                    reason="requires Windows")
+def test_windows_url():
+    assert Path(r'Foo\Bar').url() == 'file:///Foo/Bar'
+    assert Path(r'C:/Foo/Bar').url() == 'file:///C://Foo/Bar'
```

Comment:
I don't understand the motivation for the extra slash here. That seems buggy to me. I'm tempted to punt on pathname2url. Although it does appear to be producing URLs that are openable by urlopen, it doesn't seem to be producing URLs that are consistent across platforms or conforming to the RFC. For example, "Foo\Bar" should resolve to something different than "\Foo\Bar".